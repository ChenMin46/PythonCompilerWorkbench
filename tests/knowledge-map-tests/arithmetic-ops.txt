4 / 6.0
"foo" + "bar"
"cat" * 5
x + y
x * y + z
(x + 5) * (y + z)
(one * (two + 3))

# some multiline action
"foo" + \
        "bar"

(  one *
    (two 
   
  
  +
 

3))
