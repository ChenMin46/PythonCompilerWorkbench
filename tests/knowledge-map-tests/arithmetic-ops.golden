4 / 6.0
"foo" + "bar"
"cat" * 5
x + y
x * y + z
(x + 5) * (y + z)
(one * (two + 3))

# some multiline action
"foo" + \
        "bar"

(  one *
    (two 
   
  
  +
 

3))
===
Module
  body: [
    Expr
      value: BinOp >> '4 / 6.0'
        left: Num >> '4'
          n: 4
        op: Div
        right: Num >> '6.0'
          n: 6.0
    Expr
      value: BinOp >> '"foo" + "bar"'
        left: Str >> '"foo"'
          s: 'foo'
        op: Add
        right: Str >> '"bar"'
          s: 'bar'
    Expr
      value: BinOp >> '"cat" * 5'
        left: Str >> '"cat"'
          s: 'cat'
        op: Mult
        right: Num >> '5'
          n: 5
    Expr
      value: BinOp >> 'x + y'
        left: Name >> 'x'
          id: 'x'
          ctx: Load
        op: Add
        right: Name >> 'y'
          id: 'y'
          ctx: Load
    Expr
      value: BinOp >> 'x * y + z'
        left: BinOp >> 'x * y'
          left: Name >> 'x'
            id: 'x'
            ctx: Load
          op: Mult
          right: Name >> 'y'
            id: 'y'
            ctx: Load
        op: Add
        right: Name >> 'z'
          id: 'z'
          ctx: Load
    Expr
      value: BinOp >> '(x + 5) * (y + z'
        left: BinOp >> 'x + 5'
          left: Name >> 'x'
            id: 'x'
            ctx: Load
          op: Add
          right: Num >> '5'
            n: 5
        op: Mult
        right: BinOp >> 'y + z'
          left: Name >> 'y'
            id: 'y'
            ctx: Load
          op: Add
          right: Name >> 'z'
            id: 'z'
            ctx: Load
    Expr
      value: BinOp >> 'one * (two + 3'
        left: Name >> 'one'
          id: 'one'
          ctx: Load
        op: Mult
        right: BinOp >> 'two + 3'
          left: Name >> 'two'
            id: 'two'
            ctx: Load
          op: Add
          right: Num >> '3'
            n: 3
    Expr
      value: BinOp ML>> '"foo" + \\\n        "bar"'
        left: Str >> '"foo"'
          s: 'foo'
        op: Add
        right: Str >> '"bar"'
          s: 'bar'
    Expr
      value: BinOp ML>> 'one *\n    (two \n   \n  \n  +\n \n\n3'
        left: Name >> 'one'
          id: 'one'
          ctx: Load
        op: Mult
        right: BinOp ML>> 'two \n   \n  \n  +\n \n\n3'
          left: Name >> 'two'
            id: 'two'
            ctx: Load
          op: Add
          right: Num >> '3'
            n: 3
  ]
--- JSON ---
{
  "contents": [
    {
      "contents": [
        {
          "contents": [
            {
              "contents": [
                {
                  "contents": [
                    "4"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }, 
                " / ", 
                {
                  "contents": [
                    "6.0"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }
              ], 
              "id": "id_0", 
              "name": "BinOp"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "contents": [
                    "\"foo\""
                  ], 
                  "id": "id_0", 
                  "name": "Str"
                }, 
                " + ", 
                {
                  "contents": [
                    "\"bar\""
                  ], 
                  "id": "id_0", 
                  "name": "Str"
                }
              ], 
              "id": "id_0", 
              "name": "BinOp"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "contents": [
                    "\"cat\""
                  ], 
                  "id": "id_0", 
                  "name": "Str"
                }, 
                " * ", 
                {
                  "contents": [
                    "5"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }
              ], 
              "id": "id_0", 
              "name": "BinOp"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "contents": [
                    "x"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                " + ", 
                {
                  "contents": [
                    "y"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }
              ], 
              "id": "id_0", 
              "name": "BinOp"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "contents": [
                    {
                      "contents": [
                        "x"
                      ], 
                      "id": "id_0", 
                      "name": "Name"
                    }, 
                    " * ", 
                    {
                      "contents": [
                        "y"
                      ], 
                      "id": "id_0", 
                      "name": "Name"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "BinOp"
                }, 
                " + ", 
                {
                  "contents": [
                    "z"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }
              ], 
              "id": "id_0", 
              "name": "BinOp"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                "(", 
                {
                  "contents": [
                    {
                      "contents": [
                        "x"
                      ], 
                      "id": "id_0", 
                      "name": "Name"
                    }, 
                    " + ", 
                    {
                      "contents": [
                        "5"
                      ], 
                      "id": "id_0", 
                      "name": "Num"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "BinOp"
                }, 
                ") * (", 
                {
                  "contents": [
                    {
                      "contents": [
                        "y"
                      ], 
                      "id": "id_0", 
                      "name": "Name"
                    }, 
                    " + ", 
                    {
                      "contents": [
                        "z"
                      ], 
                      "id": "id_0", 
                      "name": "Name"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "BinOp"
                }
              ], 
              "id": "id_0", 
              "name": "BinOp"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        ")\n(", 
        {
          "contents": [
            {
              "contents": [
                {
                  "contents": [
                    "one"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                " * (", 
                {
                  "contents": [
                    {
                      "contents": [
                        "two"
                      ], 
                      "id": "id_0", 
                      "name": "Name"
                    }, 
                    " + ", 
                    {
                      "contents": [
                        "3"
                      ], 
                      "id": "id_0", 
                      "name": "Num"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "BinOp"
                }
              ], 
              "id": "id_0", 
              "name": "BinOp"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "))\n\n# some multiline action\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "contents": [
                    "\"foo\""
                  ], 
                  "id": "id_0", 
                  "name": "Str"
                }, 
                " + \\\n        ", 
                {
                  "contents": [
                    "\"bar\""
                  ], 
                  "id": "id_0", 
                  "name": "Str"
                }
              ], 
              "id": "id_0", 
              "name": "BinOp"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n\n(  ", 
        {
          "contents": [
            {
              "contents": [
                {
                  "contents": [
                    "one"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                " *\n    (", 
                {
                  "contents": [
                    {
                      "contents": [
                        "two"
                      ], 
                      "id": "id_0", 
                      "name": "Name"
                    }, 
                    " \n   \n  \n  +\n \n\n", 
                    {
                      "contents": [
                        "3"
                      ], 
                      "id": "id_0", 
                      "name": "Num"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "BinOp"
                }
              ], 
              "id": "id_0", 
              "name": "BinOp"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }
      ], 
      "id": "id_0", 
      "name": "Module"
    }, 
    "))\n"
  ], 
  "id": "id_ROOT", 
  "name": "ROOT_NODE"
}
